### PR TITLE
Add support for isDefault, isArchived, and refrenceNumber to Fund API

### LINKED ISSUE #3972: Add support for isDefault, isArchived, and refrenceNumber to Fund API
The frontend allows users to set a refrenceNumber, mark a fund as isDefault, and toggle isArchived when creating or updating a fund, however, these fields are currently not supported by the backend schema or mutations and attempting to include these fields in the createFund or updateFund mutation results in a GraphQL validation error.

We have to update the Fund type in the schema to include:
isDefault: Boolean
isArchived: Boolean
refrenceNumber: String

Screenshot:

Image

Related issue:

https://github.com/PalisadoesFoundation/talawa-admin/issues/5029

### PR DESCRIPTION
**Pull request by satyanvm**
**What kind of change does this PR introduce?**

This PR adds isDefault, isArchived, and refrenceNumber fiels to Fund API

**Issue Number:**

Fixes #3972 

**Snapshots/Videos:**

N/A

**If relevant, did you update the documentation?**

N/A

**Summary**

Add missing fiels to Fund API.
https://github.com/PalisadoesFoundation/talawa-api/issues/3972

**Does this PR introduce a breaking change?**

N/A

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass

**Other information**

Fund API schema addition.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

* **New Features**
  * Funds can be marked as default, archived, and carry a reference number.
  * Tag management: assign and unassign tags for users.
  * Authentication: refresh token support and related session operations.
  * New community query and expanded GraphQL responses exposing the above fields.

* **Tests**
  * Comprehensive tests for creating and updating funds (defaults, archival, reference numbers).

* **Chores**
  * Database migration and migration journal entry to add new fund fields.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by palisadoes** [comment]
@coderabbitai full review

Perform a comprehensive analysis and reevaluate all resolved items with the following criteria:

**Test Coverage Analysis:**
1. For each modified or new test file, verify:
   - All edge cases and use cases are covered (success paths, error paths, boundary conditions)
   - Test assertions are correct and meaningful
   - 100% code coverage is achieved for the associated implementation files
   - No redundant or duplicate test cases exist
   
2. For each modified or new source file, identify:
   - Specific lines of code lacking test coverage
   - Functions, branches, or conditions that need additional test cases
   - Mock/stub requirements that may be missing

**Code Quality Assessment:**
3. Identify any files that should be excluded:
   - Accidentally committed files (IDE configs, env files, temporary files)
   - Files unrelated to the PR scope
   - Generated files that should be in .gitignore

**Requirement Validation:**
4. Verify the PR fully addresses the linked issue:
   - All acceptance criteria from the issue are met
   - Implementation matches the described solution approach
   - No scope creep or unrelated changes are included

5. Confirm compliance with all previous review feedback:
   - All resolved comments have been properly addressed
   - No regressions of previously suggested improvements
   - Coding standards and conventions are followed

**Review Decision:**
- Request changes (CHANGES_REQUESTED) if ANY of the following apply:
  - Missing test coverage or invalid tests
  - Code quality issues, security concerns, or bugs
  - Incomplete implementation of issue requirements
  - Non-compliance with previous feedback
  - Presence of unnecessary files

- Approve (APPROVED) only if:
  - All criteria above are fully satisfied
  - No changes, even trivial ones, are required

**Comment by palisadoes** [comment]
@coderabbitai full review using this prompt: 
- https://github.com/PalisadoesFoundation/talawa-api/pull/4004#issuecomment-3662425246

**Comment by palisadoes** [comment]
@coderabbitai full review using this prompt: 
- https://github.com/PalisadoesFoundation/talawa-api/pull/4004#issuecomment-3662425246

File: test/graphql/types/Mutation/createFund.test.ts [hunk 1]
```diff
@@ -0,0 +1,593 @@
...
+			// Cleanup org
+			try {
+				await mercuriusClient.mutate(Mutation_deleteOrganization, {
+					headers: { authorization: `bearer ${adminAuthToken}` },
+					variables: { input: { id: orgId } },
+				});
+			} catch {
+				/* ignore */
+			}
+		});
```
[the remark this answers is not in the document]
Comment (satyanvm) [reply test/graphql/types/Mutation/createFund.test.ts:473]: For the current validation rules, this seems correct.

**Comment by satyanvm** [comment]
@coderabbitai full review using this prompt:
  https://github.com/PalisadoesFoundation/talawa-api/pull/4004#issuecomment-3662425246

**Comment by palisadoes** [comment]
@coderabbitai full review

Perform a comprehensive analysis and reevaluate all resolved items with the following criteria:

**Test Coverage Analysis:**
1. For each modified or new test file, verify:
   - All edge cases and use cases are covered (success paths, error paths, boundary conditions)
   - Test assertions are correct and meaningful
   - 100% code coverage is achieved for the associated implementation files
   - No redundant or duplicate test cases exist
   
2. For each modified or new source file, identify:
   - Specific lines of code lacking test coverage
   - Functions, branches, or conditions that need additional test cases
   - Mock/stub requirements that may be missing

**Code Quality Assessment:**
3. Identify any files that should be excluded:
   - Accidentally committed files (IDE configs, env files, temporary files)
   - Files unrelated to the PR scope
   - Generated files that should be in .gitignore

**Requirement Validation:**
4. Verify the PR fully addresses the linked issue:
   - All acceptance criteria from the issue are met
   - Implementation matches the described solution approach
   - No scope creep or unrelated changes are included

5. Confirm compliance with all previous review feedback:
   - All resolved comments have been properly addressed
   - No regressions of previously suggested improvements
   - Coding standards and conventions are followed

**Review Decision:**
- Request changes (CHANGES_REQUESTED) if ANY of the following apply:
  - Missing test coverage or invalid tests
  - Code quality issues, security concerns, or bugs
  - Incomplete implementation of issue requirements
  - Non-compliance with previous feedback
  - Presence of unnecessary files

- Approve (APPROVED) only if:
  - All criteria above are fully satisfied
  - No changes, even trivial ones, are required

**Review by palisadoes — APPROVED** [review]
